test/test.js

```diff
@@ -1,193 +1,185 @@
 'use strict';
 
-const test = require('tap').test;
-
+const expect = require('expect');
+const fs = require('fs');
 const gulp = require('gulp');
-const task = require('../');
+const { concat, pipe } = require('mississippi');
+const path = require('path');
 const pug = require('pug');
 const through = require('through2');
-const path = require('path');
-const fs = require('fs');
-const extname = require('path').extname;
+const task = require('../');
 
 const filename = path.join(__dirname, './fixtures/helloworld.pug');
 
 // Mock Data Plugin
 // (not testing the gulp-data plugin options, just that gulp-pug can get its
 // data from file.data)
-const setData = function setData() {
-  return through.obj(function(file, enc, callback) {
+function setData() {
+  return through.obj((file, _enc, cb) => {
     file.data = {
       title: 'Greetings!',
     };
-    // eslint-disable-next-line no-invalid-this
-    this.push(file);
-    return callback();
+    cb(null, file);
   });
-};
+}
 
-const expectStream = function expectStream(t, options, finish) {
+function expectStream(options) {
   options = options || {};
-  if (typeof finish === 'undefined') {
-    finish = true;
-  }
   const ext = options.client ? '.js' : '.html';
   const compiler = options.client ? pug.compileClient : pug.compile;
-  return through.obj(function expectData(file, enc, cb) {
+
+  function assert(file, _enc, cb) {
     options.filename = file.path.replace(new RegExp(ext + '$'), '.pug');
     const compiled = compiler(fs.readFileSync(options.filename), options);
     const data = Object.assign({}, options.data, options.locals, file.data);
     const expected = options.client ? compiled : compiled(data);
-    t.equal(expected, String(file.contents));
-    t.equal(extname(file.path), ext);
+    expect(file.contents.toString()).toStrictEqual(expected);
+    expect(path.extname(file.path)).toStrictEqual(ext);
     if (file.relative) {
-      t.equal(extname(file.relative), ext);
+      expect(path.extname(file.relative)).toStrictEqual(ext);
     } else {
-      t.equal(extname(file.relative), '');
+      expect(path.extname(file.relative)).toStrictEqual('');
     }
-    if (finish) {
-      t.end();
-    }
-    // eslint-disable-next-line no-invalid-this
-    this.push(file);
-    cb();
-  });
-};
+    cb(null, file);
+  }
 
-test('should compile my pug files into HTML', function(t) {
-  gulp.src(filename)
-    .pipe(task())
-    .pipe(expectStream(t));
-});
+  return through.obj(assert);
+}
 
-test('should compile my pug files into HTML with locals', function(t) {
-  gulp.src(filename)
-    .pipe(task({
-      locals: {
-        title: 'Yellow Curled',
-      },
-    }))
-    .pipe(expectStream(t, {
+describe('test', function() {
+  it('should compile my pug files into HTML', function(done) {
+    pipe([
+      gulp.src(filename),
```

Comment:
I'd prefer to mock all gulp.src calls (as `from.obj([file])`) so we don't have a dependency on gulp.

Reply:
I dropped the gulp dev-dependency (pushed to this branch).